lib/stripe_mock/request_handlers/subscriptions.rb

```diff
@@ -137,6 +137,21 @@ def create_subscription(route, method_url, params, headers)
           subscription[:canceled_at] = Time.now.utc.to_i
         end
 
+        if (s = params[:expand]&.first { |s| s.starts_with? 'latest_invoice' })
+          intent = Data.mock_payment_intent({
+            status: 'succeeded',
```

Comment:
Subscriptions may be created as "incomplete", so I think you need to set the status of "payment_intent" based on the "status" instead of always setting it to "success".

when Stripe::Subscription was created as "imcomplete", the "payment_intent"'s stauts is "requires_payment_method".

https://github.com/stripe-ruby-mock/stripe-ruby-mock/pull/801